hathor/builder.py

```diff
@@ -36,12 +40,308 @@
 from hathor.p2p.utils import discover_hostname
 from hathor.prometheus import PrometheusMetricsExporter
 from hathor.pubsub import PubSubManager
+from hathor.storage import RocksDBStorage
+from hathor.transaction.storage import TransactionMemoryStorage, TransactionRocksDBStorage, TransactionStorage
+from hathor.util import Random, Reactor, get_environment_info
 from hathor.wallet import BaseWallet, HDWallet, Wallet
 
 logger = get_logger()
 
 
+class StorageType(Enum):
+    MEMORY = 'memory'
+    ROCKSDB = 'rocksdb'
+
+
+class BuildArtifacts(NamedTuple):
+    """Artifacts created by a builder."""
+    settings: HathorSettingsType
+    rng: Random
+    reactor: Reactor
+    manager: HathorManager
+    pubsub: PubSubManager
+    consensus: ConsensusAlgorithm
+    tx_storage: TransactionStorage
+    indexes: Optional[IndexesManager]
+    wallet: Optional[BaseWallet]
+    rocksdb_storage: Optional[RocksDBStorage]
+
+
+class Builder:
```

Comment:
What do you think about moving this and other new related classes to their own file?

Reply:
Sure! It makes a lot of sense. Thanks! Done!